tests/regression/test_mesh_vertexonly.py

```diff
@@ -0,0 +1,333 @@
+        swarm.restoreField(fieldname)
+    # Check comm sizes match
+    assert plex.comm.size == swarm.comm.size
+    # Check coordinate list and parent cell indices match
+    assert len(localpointcoords) == len(localparentcellindices)
+    # check local points are found in list of input points
+    for p in localpointcoords:
+        assert np.any(np.isclose(p, inputpointcoords))
+    # check local points are correct local points given mesh
+    # partitioning (but don't require ordering to be maintained)
+    assert len(localpointcoords) == len(inputlocalpointcoords)
+    assert np.all(np.isin(inputlocalpointcoords, localpointcoords))
+    # Check methods for checking number of points on current MPI rank
+    assert len(localpointcoords) == swarm.getLocalSize()
+    # Check there are as many local points as there are local cells
+    # (excluding ghost cells in the halo)
+    assert len(localpointcoords) == parentmesh.cell_set.size
+    # Check total number of points on all MPI ranks is correct
+    # (excluding ghost cells in the halo)
+    assert nptsglobal == len(inputpointcoords)
+    assert nptsglobal == swarm.getSize()
+    # Check the parent cell indexes match those in the parent mesh
+    cell_indexes = parentmesh.cell_closure[:, -1]
+    for index in localparentcellindices:
+        assert np.any(index == cell_indexes)
+
+
+@pytest.mark.parallel
+@pytest.mark.parametrize("parentmesh", parentmeshes)
+def test_pic_swarm_in_plex_parallel(parentmesh):
+    test_pic_swarm_in_plex(parentmesh)
+
+
+@pytest.mark.parallel(nprocs=2)  # nprocs == total number of mesh cells
+def test_pic_swarm_in_plex_2d_2procs():
+    test_pic_swarm_in_plex(UnitSquareMesh(1, 1))
+
+
+@pytest.mark.parallel(nprocs=3)  # nprocs > total number of mesh cells
+def test_pic_swarm_in_plex_2d_3procs():
+    test_pic_swarm_in_plex(UnitSquareMesh(1, 1))
+
+
+# Mesh Generation Tests
+
+def verify_vertexonly_mesh(m, vm, inputvertexcoords, gdim):
+    """
+    Check that VertexOnlyMesh `vm` immersed in parent mesh `v` with
+    creation coordinates `inputvertexcoords` and geometric dimension
+    `gdim` behaves as expected. `inputvertexcoords` should be the same
+    for all MPI ranks to avoid hanging.
+    """
+    assert m.geometric_dimension() == gdim
+    # Correct dims
+    assert vm.geometric_dimension() == gdim
+    assert vm.topological_dimension() == 0
+    # Can initialise
+    vm.init()
+    # Find in-bounds and non-halo-region input coordinates
+    in_bounds = []
+    core, owned, ghost = m.cell_set.sizes
+    for i in range(len(inputvertexcoords)):
+        cell_num = m.locate_cell(inputvertexcoords[i])
+        if cell_num is not None and cell_num < owned:
+            in_bounds.append(i)
+    # Correct coordinates (though not guaranteed to be in same order)
+    assert np.shape(vm.coordinates.dat.data_ro) == np.shape(inputvertexcoords[in_bounds])
+    assert np.all(np.isin(inputvertexcoords[in_bounds], vm.coordinates.dat.data_ro))
```

Comment:
This is quadratic, right? Sort and then say `np.allclose(...)`?

Reply:
I don't know what you mean? Sorting and using `np.allclose` does make more sense though.